### ISSUE TITLE
Events may be called twice when using `movesPerTurn`

### ISSUE THREAD

**Issue by Stefan-Hanke** [nicolodavis/boardgame.io#242]
This is a follow-up issue from #241.

When a flow uses the `movesPerTurn` property (e.g. in the [TicTacToe example](https://github.com/google/boardgame.io/blob/b9d703832cf37d78d25c6208338e6d2bd7c01df1/examples/react/modules/tic-tac-toe/game.js#L58)), the library will call events automatically, even when a move also called into events.

Events should be called exactly once as they happen, regardless of whether a game calls into events or not.

**Comment by nicolodavis** [comment]
Should we just consider this user error, i.e. if you use `movesPerTurn`, your game logic is flawed if you also call `endTurn` from your move?

What about:
```
move: moveName(G, ctx) {
  ctx.events.endTurn();
  ctx.events.endTurn();
}
```

This will also result in multiple `endTurn` calls being emitted.  Should we try to second guess this as well?

The reasons I'm OK with writing this off as user error without writing more code to handle it:

1. The turn can only be ended exactly once.  The following snippet logs "turn end" only once despite the fact that multiple end turn events are emitted.

```
move: moveName(G, ctx) {
  ctx.events.endTurn();
  ctx.events.endTurn();
},

flow: {
  onTurnEnd(G) {
    console.log('turn end');
    return G;
  }
}
```

2. You can only end your own turn.  There isn't a danger of a stray `endTurn` affecting another player's turn. Reference: https://github.com/google/boardgame.io/commit/c4a11a79292700cf9d6afde737cabd0089941b8d

**Comment by Stefan-Hanke** [comment]
Sounds reasonable. 

I don't know whether this is documented, but there should be a paragraph about using `movesPerTurn` in a flow.

What I'm unhappy with is that such <s>errors</s> behaviours get silently ignored (e.g. it's not allowed to call that event? The framework will just ignore it...). The game should work given all those checks but I'd like it to be a bit more perceptible.
